Feat: add diesel-rs

JedBeom opened the pull request:
- Remove sqlx
- Install diesel-rs and deadpool-diesel
- Re-write migration files
- Add mods `schema` and `models`
- Specify `toolchain: nightly` in `backend.yaml`

## Test Plan

- Install postgresql and edit `DATABASE_URL` in `.env`
- Run the server
- Send **POST** http://localhost:8080/test/create_configs
- GET http://localhost:8080/test/get_configs

chocological00 commented:
format plz

chocological00 commented on migrations/2025-01-31-152213_init_stage01/up.sql:
```diff
@@ -0,0 +1,45 @@
+-- Your SQL goes here
+CREATE TABLE "users"(
+	"id" BIGSERIAL NOT NULL PRIMARY KEY,
+	"username" VARCHAR NOT NULL UNIQUE,
```
limit 255?

chocological00 commented on Cargo.toml:
```diff
@@ -15,19 +15,14 @@ dotenvy = "0.15.7"
 tracing = "0.1.41"
 tracing-subscriber = { version = "0.3.19", features = ["env-filter"] }
 reqwest = { version = "0.12.12", features = ["stream"] }
+diesel = { version = "2.2.7", features = ["postgres", "chrono"] }
```
swap `s/chrono/extras/` - see https://docs.rs/crate/diesel/latest/features#extras

chocological00 commented on src/db/mod.rs:
```diff
@@ -1,29 +1,12 @@
-        .after_connect(|_conn, _meta| Box::pin(async move { Ok(()) }))
-        .connect_with(conn_opts)
-        .await
+pub async fn init(database_url: &str) -> Result<Pool, BuildError> {
```
drop async?

JedBeom commented:
<img width="374" alt="Screenshot 2025-02-20 at 21 21 49" src="https://github.com/user-attachments/assets/b42bc3fc-cede-42b7-b36d-e0b7b4e7511a" />

`cargo fmt` does not occur any errors or changes on my machine :( strange

JedBeom committed:
Feat: add diesel-rs

- Remove sqlx
- Install diesel-rs and deadpool-diesel
- Re-write migration files
- Add mods `schema` and `models`
- Specify `toolchain: nightly` in `backend.yaml`

## Test Plan

- Install postgresql and edit `DATABASE_URL` in `.env`
- Run the server
- Send **POST** http://localhost:8080/test/create_configs
- GET http://localhost:8080/test/get_configs
